### PR TITLE
added a better url joiner

### PR DESCRIPTION
**Pull request by jdamick**

### CODE REVIEW

File: jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java [hunk 1]
```diff
@@ -304,12 +303,24 @@ protected HttpHeaders createHeaders(final String userInfo) {
...
+            return makeRestCall(joinURL(buildUrl, "consoleText")).getBody();
         } catch (MalformedURLException mfe) {
             LOG.error("malformed url for build log", mfe);
         }
 
         return "";
     }
+
+    // join a base url to another path or paths - this will handle trailing or non-trailing /'s
+    public static String joinURL(String base, String ... paths) throws MalformedURLException {
```
Comment (tabladrum) [inline jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java]: Move this to a base class or Utility class?

Comment (jdamick) [inline jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java]: i was thinking about creating a core-collector or something along those lines to consolidate all of these

File: jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java [hunk 2]
```diff
@@ -304,12 +303,24 @@ protected HttpHeaders createHeaders(final String userInfo) {
 
     protected String getLog(String buildUrl) {
         try {
-            String fullUrl = (new URL(URI.create(buildUrl).toURL(), "consoleText")).toString();
-            return makeRestCall(fullUrl).getBody();
+            return makeRestCall(joinURL(buildUrl, "consoleText")).getBody();
         } catch (MalformedURLException mfe) {
```
Comment (amitmawkin) [inline jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java]: we are eating a runtime Exception ???

Comment (jdamick) [inline jenkins-build-collector/src/main/java/com/capitalone/dashboard/collector/DefaultHudsonClient.java]: apparently - this is existing cruft

**Pull request merged**